src/common/s3.connection.hh

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <condition_variable>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace minio::s3 {
+class Client;
```

Comment:
It was meant to be an optimization and also not to clutter up the namespace, because eventually s3.connection.hh will be `#include`d in s3.sink.hh which will be `#include`d in lots of other files, but it's probably not a big deal.

Reply:
I noticed you kept it as is. It's not a blocker, but I think it's better to avoid premature optimizations if it's not a big deal.

Reply:
No, I just forgot to do that.